src/windows/mod.rs

```diff
@@ -170,7 +178,164 @@ impl GeneralReadout for WindowsGeneralReadout {
     }
 
     fn resolution(&self) -> Result<String, ReadoutError> {
-        Err(ReadoutError::NotImplemented)
+        let mut devices = Vec::new();
+        let mut index = 0;
+        let mut status = true;
+        // Iterate over EnumDisplayDevicesW until it returns false
+        while status {
+            devices.push(DISPLAY_DEVICEW::default());
+            devices[index].cb = std::mem::size_of::<DISPLAY_DEVICEW>() as u32;
+            status = unsafe {
+                EnumDisplayDevicesW(
+                    PCWSTR::null(),
+                    index as u32,
+                    &mut devices[index],
+                    EDD_GET_DEVICE_INTERFACE_NAME,
+                )
+                .as_bool()
+            };
+            index += 1;
+        }
+        // Remove the last element, which will be invalid
+        devices.pop();
+
+        // Create a vector to store each monitor's information
+        let mut resolutions = Vec::new();
+
+        // Iterate over each device
+        let mut index = 0;
+        for device in devices {
+            resolutions.push(DEVMODEW::default());
+            resolutions[index].dmSize = std::mem::size_of::<DEVMODEW>() as u16;
+
+            // Get the current display settings for the device
+            unsafe {
+                EnumDisplaySettingsW(
+                    PCWSTR(device.DeviceName.as_ptr()),
+                    ENUM_CURRENT_SETTINGS,
+                    &mut resolutions[index],
+                );
+            }
+
+            // Ensure that the resolution is valid
+            if resolutions[index].dmPelsWidth != 0 && resolutions[index].dmPelsHeight != 0 {
+                index += 1;
+            } else {
+                resolutions.pop();
+            }
+        }
+
+        if !resolutions.is_empty() {
+            // Format and return the display resolutions and refresh rates
+            return Ok(resolutions
+                .iter()
+                .map(
+                    |resolution| match (resolution.dmDisplayFrequency, resolution.dmLogPixels) {
```

Comment:
Did you try updating the scale factor of your monitors? `dmLogPixels` won't be updated before I restarting my laptap.

Reply:
> Did you try updating the scale factor of your monitors? For me, `dmLogPixels` won't be updated before I restarting my laptap.
> 
> In addition, according to MSDN, `EnumDisplaySettings` sets only these 5 DEMMODE members.
> 
> ![image](https://user-images.githubusercontent.com/6134068/234348148-1c6c2c03-9c8b-42e9-9f49-563b0043e915.png)

I didn't before now, and I can confirm that the issue exists for me as well. Along with that, I found that the current implementation doesn't properly get the scale for a second monitor even after a restart. I see a few options moving forward:
1. Only show the scaled resolution of the primary monitor and have the bug where it doesn't update as a known issue
2. Don't show a scaled resolution
3. Do [what Fastfetch does](https://github.com/CarterLi/fastfetch/blob/dev/src/detection/displayserver/displayserver_windows.c#L62) and merge the two implementations into a single one that uses `EnumDisplaySettingsW` to get the resolution and refresh rate and uses `EnumDisplayMonitors` without `SetProcessDpiAwareness` to get the scaled resolution. This would likely be about twice as slow, but would add a maximum of a few milliseconds.

I'm going to try implementing option 3 and see how much slower it is.

> Did you test multiple monitors?

When a monitor is added or removed, the output correctly updates

Reply:
There is actually another method by using HDC and GetDeviceCaps.

https://github.com/CarterLi/fastfetch/commit/4d7cb25aa61e5cd06779e126046267edb062cd49#diff-1f993416a36137ee7af25d94be4928701d89ca08d9ab322fcb3eb0b5ec6d80b6R39

I dont know if it is faster than EnumDisplayMonitors. The code is much simplier at least.